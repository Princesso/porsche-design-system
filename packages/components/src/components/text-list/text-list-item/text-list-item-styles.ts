import type { JssStyle } from 'jss';
import type { ListType, OrderType } from '../text-list/text-list-utils';
import { getCss } from '../../../utils';
import { addImportantToEachRule, pxToRemWithUnit } from '../../../styles';
import { textSmall } from '@porsche-design-system/utilities-v2';

const cssVariableOrderedSuffix = '--pds-text-list-item-ordered-suffix';
const cssVariableUnorderedWidth = '--pds-text-list-item-unordered-width';
const cssVariableUnorderedHeight = '--pds-text-list-item-unordered-height';
const cssVariableUnorderedTop = '--pds-text-list-item-unordered-top';

export const getComponentCss = (listType: ListType, orderType: OrderType): string => {
  const isOrdered = listType === 'ordered';
  const beforeJssStyles: JssStyle = {
    position: 'absolute',
    left: 0,
  };

  return getCss({
    '@global': {
      '::slotted(*)': {
        [cssVariableOrderedSuffix]: '""',
        [cssVariableUnorderedWidth]: pxToRemWithUnit(8),
        [cssVariableUnorderedHeight]: '1px',
        [cssVariableUnorderedTop]: 'calc(1.5em / 2)',
      },
      ':host': addImportantToEachRule({
        position: 'relative',
        display: 'list-item',
        color: 'inherit',
        listStyleType: 'none',
        paddingLeft: pxToRemWithUnit(isOrdered ? 40 : 24),
        '&:before': isOrdered
          ? {
              ...beforeJssStyles,
              content: `counters(section,".",${
                orderType === 'numbered' ? 'decimal' : 'lower-latin'
              }) var(${cssVariableOrderedSuffix},".")`,
              top: 0,
              width: pxToRemWithUnit(24),
              height: 'auto',
              counterIncrement: 'section',
              textAlign: 'right',
              backgroundColor: 'transparent',
              ...textSmall,
            }
          : {
              ...beforeJssStyles,
              content: '""',
              width: `var(${cssVariableUnorderedWidth},${pxToRemWithUnit(4)})`,
              height: `var(${cssVariableUnorderedHeight},${pxToRemWithUnit(4)})`,
              top: `var(${cssVariableUnorderedTop},calc(1.5em / 2 - 0.125em))`,
              backgroundColor: 'currentColor',
            },
      }),
    },
  });
};
