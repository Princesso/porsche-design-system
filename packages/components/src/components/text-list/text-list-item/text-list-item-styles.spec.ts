import { getComponentCss } from './text-list-item-styles';

xdescribe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([
    ['unordered', 'numbered'],
    ['unordered', 'alphabetically'],
    ['ordered', 'numbered'],
    ['ordered', 'alphabetically'],
  ])('should return correct css for listType: %s and orderType: %s', (...args) => {
    expect(getComponentCss(...args)).toMatchSnapshot();
  });
});
