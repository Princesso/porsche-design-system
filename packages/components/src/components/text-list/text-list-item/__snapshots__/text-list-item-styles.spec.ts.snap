// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css for listType: ordered and orderType: alphabetically 1`] = `
"::slotted(*) {
  --pds-text-list-item-ordered-suffix: "";
  --pds-text-list-item-unordered-width: 0.5rem;
  --pds-text-list-item-unordered-height: 1px;
  --pds-text-list-item-unordered-top: calc(1.5em / 2);
}
:host {
  position: relative !important;
  display: list-item !important;
  color: inherit !important;
  list-style-type: none !important;
  padding-left: 2.5rem !important;
}
:host:before {
  position: absolute !important;
  left: 0 !important;
  content: counters(section,".",lower-latin) var(--pds-text-list-item-ordered-suffix,".") !important;
  top: 0 !important;
  width: 1.5rem !important;
  height: auto !important;
  counter-increment: section !important;
  text-align: right !important;
  background-color: transparent !important;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif !important;
  text-size-adjust: none !important;
  -webkit-text-size-adjust: none !important;
  overflow-wrap: break-word !important;
  hyphens: auto !important;
}"
`;

exports[`getComponentCss() should return correct css for listType: ordered and orderType: numbered 1`] = `
"::slotted(*) {
  --pds-text-list-item-ordered-suffix: "";
  --pds-text-list-item-unordered-width: 0.5rem;
  --pds-text-list-item-unordered-height: 1px;
  --pds-text-list-item-unordered-top: calc(1.5em / 2);
}
:host {
  position: relative !important;
  display: list-item !important;
  color: inherit !important;
  list-style-type: none !important;
  padding-left: 2.5rem !important;
}
:host:before {
  position: absolute !important;
  left: 0 !important;
  content: counters(section,".",decimal) var(--pds-text-list-item-ordered-suffix,".") !important;
  top: 0 !important;
  width: 1.5rem !important;
  height: auto !important;
  counter-increment: section !important;
  text-align: right !important;
  background-color: transparent !important;
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif !important;
  text-size-adjust: none !important;
  -webkit-text-size-adjust: none !important;
  overflow-wrap: break-word !important;
  hyphens: auto !important;
}"
`;

exports[`getComponentCss() should return correct css for listType: unordered and orderType: alphabetically 1`] = `
"::slotted(*) {
  --pds-text-list-item-ordered-suffix: "";
  --pds-text-list-item-unordered-width: 0.5rem;
  --pds-text-list-item-unordered-height: 1px;
  --pds-text-list-item-unordered-top: calc(1.5em / 2);
}
:host {
  position: relative !important;
  display: list-item !important;
  color: inherit !important;
  list-style-type: none !important;
  padding-left: 1.5rem !important;
}
:host:before {
  position: absolute !important;
  left: 0 !important;
  content: "" !important;
  width: var(--pds-text-list-item-unordered-width,0.25rem) !important;
  height: var(--pds-text-list-item-unordered-height,0.25rem) !important;
  top: var(--pds-text-list-item-unordered-top,calc(1.5em / 2 - 0.125em)) !important;
  background-color: currentColor !important;
}"
`;

exports[`getComponentCss() should return correct css for listType: unordered and orderType: numbered 1`] = `
"::slotted(*) {
  --pds-text-list-item-ordered-suffix: "";
  --pds-text-list-item-unordered-width: 0.5rem;
  --pds-text-list-item-unordered-height: 1px;
  --pds-text-list-item-unordered-top: calc(1.5em / 2);
}
:host {
  position: relative !important;
  display: list-item !important;
  color: inherit !important;
  list-style-type: none !important;
  padding-left: 1.5rem !important;
}
:host:before {
  position: absolute !important;
  left: 0 !important;
  content: "" !important;
  width: var(--pds-text-list-item-unordered-width,0.25rem) !important;
  height: var(--pds-text-list-item-unordered-height,0.25rem) !important;
  top: var(--pds-text-list-item-unordered-top,calc(1.5em / 2 - 0.125em)) !important;
  background-color: currentColor !important;
}"
`;
